src/VCR_SystemTasks.cpp

```diff
@@ -0,0 +1,52 @@
+#include "etl/delegate.h"
+#include "FlexCAN_T4.h"
+
+
+#include "SharedFirmwareTypes.h"
+#include "CANInterface.h"
+
+#include "VCR_SystemTasks.h"
+#include "VCFInterface.h"
+#include "VCRCANInterfaceImpl.h"
+#include "VehicleStateMachine.h"
+
+auto recv_call = etl::delegate<void(CANInterfaces& , const CAN_message_t& , unsigned long )>::create<VCRCANInterfaceImpl::vcr_CAN_recv>();
+
+InterfaceData_s sample_interfaces(unsigned long curr_millis, VCRInterfaces & interface_ref_container)
+{
+    // process ring buffer is from CANInterface. TODO put into namespace
+    process_ring_buffer(CAN2_rxBuffer, interface_ref_container.can_interfaces, curr_millis, recv_call);
+    process_ring_buffer(CAN3_rxBuffer, interface_ref_container.can_interfaces, curr_millis, recv_call);
+
+    auto current_vcf_data = interface_ref_container.can_interfaces.vcf_interface.get_latest_data();
+    InterfaceData_s interf_data = {.current_pedals_data = current_vcf_data.pedals_data.recvd_data };
```

Comment:
nit: can we refrain from using "current"? Either abbreviate it to curr, or omit it altogether to simply call it pedals_data.